src/wallet/wallet.cpp

```diff
@@ -1180,9 +1173,14 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFlushOnClose)
     return true;
 }
 
-void CWallet::LoadToWallet(const CWalletTx& wtxIn)
+void CWallet::LoadToWallet(CWalletTx& wtxIn)
 {
     uint256 hash = wtxIn.GetHash();
+    Optional<int> block_height = MakeOptional(false, int());
+    if (hasChain()) {
```

Comment:
Okay, this point is already mentioned in commit message, `Verify` and `LoadWallet` but added also on this place.

Reply:
Instead of initializing an `Optional<int>` then conditionally assigning it based on the result of `hasChain()`, could you instead define something like `chainHeight()` to return an `Optional<int>`? This would hide the chain logic from `LoadToWallet()`.

If not, consider directly assigning `0` to `wtxIn.m_block_height` and conditionally updating it based on `hasChain()`. The addition of `Optional<int>` here just seems to add indirection for the reader.

Reply:
I don't want to add new method in Chain interface, but yes 2nd suggested alternative avoid the `MakeOptional` which is good, updated